hierarchicalforecast/methods.py

```diff
@@ -966,6 +970,35 @@ def fit_predict(self,
 
     __call__ = fit_predict
 
+@njit(parallel=True, fastmath=True, error_model="numpy")
+def _ma_cov(residuals: np.ndarray, not_nan_mask: np.ndarray):
+    """Masked empirical covariance matrix.
+
+    :meta private:
+    """
+    n_timeseries = residuals.shape[0]
+    W = np.zeros((n_timeseries, n_timeseries), dtype=np.float64).T
+    for i in prange(n_timeseries):
+        not_nan_mask_i = not_nan_mask[i]
+        for j in range(i + 1):
+            not_nan_mask_j = not_nan_mask[j]
+            not_nan_mask_ij = not_nan_mask_i & not_nan_mask_j   
+            n_samples = np.sum(not_nan_mask_ij)
+            # Only compute if we have enough non-nan samples in the time series pair
+            if n_samples > 1:
+                # Masked residuals
+                residuals_i = residuals[i][not_nan_mask_ij]
+                residuals_j = residuals[j][not_nan_mask_ij]
+                residuals_i_mean = np.mean(residuals_i)
+                residuals_j_mean = np.mean(residuals_j)
+                X_i = (residuals_i - residuals_i_mean)
+                X_j = (residuals_j - residuals_j_mean)
+                # Empirical covariance
+                factor_emp_cov = np.float64(1 / (n_samples - 1))
+                W[i, j] = W[j, i] = factor_emp_cov * np.mean(X_i * X_j)
```

Comment:
```suggestion
                W[i, j] = W[j, i] = factor_emp_cov * np.sum(X_i * X_j)
            else:
                W[i, j] = W[j, i] = np.nan
```

We need the sum of the product of the deviations as we are dividing by `n_samples - 1`, and `np.nan` for the cases without enough samples.

Reply:
Thanks! I made the change for the mean, and added a test to make sure the empirical covariance is the same across methods (when calculating without NaNs)

`np.nan` for cases without enough samples is of course the technically correct method, but that will also give an error later on - I'd prefer to just set it to zero in those cases.